packages/geocoding/lib/providers/geonames.provider.ts

```diff
@@ -0,0 +1,95 @@
+import { GeocodingResult } from "../model";
+import { Geometry } from "geojson";
+
+const baseUrl = "https://secure.geonames.org/searchJSON";
+
+/**
+ * Reference documentation: http://www.geonames.org/export/geonames-search.html
+ * @property lang Place name and country name will be returned in the specified language. Default is English.
+ * @property maxRows The maximal number of rows in the document returned by the service. Default is 100
+ * @property country Default is all countries. The country parameter may occur more than once, example: country=FR&country=GP
+ * @property style verbosity of returned xml document, default = MEDIUM
+ * @property type The format type of the returned document, default = xml
+ * @property username The parameter 'username' needs to be passed with each request. (https://www.geonames.org/login)
+ * @property east, west, north, south Restrict the search to the given bounding box
+ */
+
+export interface GeonamesOptions {
+    lang?: "de" | "fr" | "it" | "rm" | "en";
+    maxRows?: number;
+    country?: string | string[]; // Now supports multiple countries
+    style?: "SHORT" | "MEDIUM" | "LONG" | "FULL";
+    type?: "json" | "xml" | "rdf";
```

Comment:
I guess this shouldn't be part of the options, since in the end we always want a `GeocodingResult` type

Reply:
@jahow  Thanks for the feedback. I understand that:

- **maxRows** helps limit search results as needed.
- **Country** allows searches to focus on specific countries.

This can be beneficial under certain situations, currently I am using maxRows to limit search to 5 results.

Agreed on removing 'style' and 'type', using defaults instead. Does this match your thoughts?

Reply:
Oh non sorry my comment was really only on the `type` options, the rest can stay :) for the `style` option I'm not sure yet but we can leave it there